contributors/design-proposals/pod-priority-api.md

```diff
@@ -0,0 +1,203 @@
+* Best effort / opportunistic
+
+### Resolving priority names 
+
+User requests sent to Kubernetes may have "PriorityName"  in their PodSpec.
+Admission controller resolves a PriorityName to its corresponding number and
+populates the "Priority" field of the pod status. The rest of Kubernetes
+components look at the "Priority" field of pod status and work with the integer
+value. In other words, "PriorityName" will be ignored by the rest of the
+system.
+
+### Modifying priority names 
+
+Admission controller can optionally receive a ConfigMap from the API server
+that defines a list of new priority names and their numbers or changes the
+integral value of existing ones. New configurations cannot introduce priority
+names mapped to a value higher than "system" and cannot change the value of
+"system" priority. The reason for this restriction is that kubernetes critical
+system pods will have "system" priority and no pod should be able to preempt
+them.
+
+Alternatively, we may ship Kubernetes with only the highest priority level
+hard-coded. The rest of the priority names are defined in a ConfigMap which
+could be shipped with Kubernetes.
+
+### Ordering of priorities 
+
+As mentioned earlier, a PriorityName is resolved by the admission controller to
+its integral value and Kubernetes components use the integral value. The higher
+the value, the higher the priority.
+
+### Dynamic change of priority names 
+
+Kubernetes supports change of priority names at runtime, however the changes
+won't change the priority of existing pods. As mentioned earlier, admission
+controller resolves priority names and populates the "Priority" field of pod
+specs. A new ConfigMap may be fed to the admission controller to define new
+priority names or change the integer value of existing names (other than
+"system"). These new changes affects only the pods created after the changes
+and existing pods will keep their existing Priority.
+
+#### Drawbacks of changing priority names 
+
+While Kubernetes allows changing priority names, it should be done only when
+absolutely needed. Changing priority names has the following disadvantages:
+
+
+
+*   May remove config portability: pod specs written for one cluster are no
+    longer guaranteed to work on a different cluster if the same priority names
+    do not exist on the new cluster. 
+*   If quota is specified for existing priorities (at the time of this writing,
+    we don't have this feature in Kubernetes), adding or changing priorities
+    will require reconfiguration of quota allocations.
+
+### Priority and QoS classes 
+
+Kubernetes has [three QoS
+classes](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-qos.md#qos-classes)
+which are derived from request and limit of pods. Priority is introduced as an
+independent concept; meaning that any QoS class may have any valid priority.
+When a node is out of resources and pods needs to be preempted, we give
```

Comment:
how about a mechanism to say that this priority can only be used with an allowed set of QoS class? i want a way to prevent what i find is an abuse vector (a BE pod using high priorities), as I think it will be a future support case when it causes an unforeseen consequence.